ext/html_tokenizer_ext/parser.c

```diff
@@ -549,6 +550,8 @@ static VALUE parser_append_data(VALUE self, VALUE source, int is_placeholder)
     parser->tk.scan.mb_cursor = mb_cursor;
 
     tokenizer_scan_all(&parser->tk);
+
+    parser->tk.scan.string = NULL;
```

Comment:
the line above does
```
parser->tk.scan.string = parser->doc.data;
```
setting this to null is intended to double free since `parser->doc.data` is also freed in `parser_free`.